velox/functions/prestosql/window/tests/RankTest.cpp

```diff
@@ -22,116 +22,85 @@ namespace facebook::velox::window::test {
 
 namespace {
 
-class RankTest : public WindowTestBase {
+static const std::vector<std::string> kRankFunctions = {
+    std::string("rank()"),
+    std::string("dense_rank()"),
+    std::string("percent_rank()"),
+    std::string("cume_dist()"),
+    std::string("row_number()")};
+
+// The WindowFunctionTest class is used to instantiate parameterized window
+// function tests. The parameters are based on the function being tested
+// and a specific over clause. The window function is tested for the over
+// clause and all combinations of frame clauses.
+struct WindowTestParam {
+  const std::string function;
+  const std::string overClause;
+};
+
+class WindowFunctionTest : public WindowTestBase {
  protected:
-  explicit RankTest(const std::string& rankFunction)
-      : rankFunction_(rankFunction) {}
+  explicit WindowFunctionTest(const WindowTestParam& testParam)
+      : function_(testParam.function), overClause_(testParam.overClause) {}
 
-  void testWindowFunction(
-      const std::vector<RowVectorPtr>& vectors,
-      const std::vector<std::string>& overClauses) {
-    WindowTestBase::testWindowFunction(vectors, rankFunction_, overClauses);
+  void testWindowFunction(const std::vector<RowVectorPtr>& vectors) {
+    WindowTestBase::testWindowFunction(vectors, function_, {overClause_});
   }
 
-  const std::string rankFunction_;
-};
-
-class MultiRankTest : public RankTest,
-                      public testing::WithParamInterface<std::string> {
- public:
-  MultiRankTest() : RankTest(GetParam()) {}
+  const std::string function_;
+  const std::string overClause_;
 };
 
-TEST_P(MultiRankTest, basic) {
-  testWindowFunction({makeSimpleVector(50)}, kBasicOverClauses);
-}
-
-TEST_P(MultiRankTest, basicWithSortOrder) {
-  testWindowFunction({makeSimpleVector(50)}, kSortOrderBasedOverClauses);
+std::vector<WindowTestParam> getWindowTestParams(
```

Comment:
This function seems to be repeated suggesting an opportunity for refactoring.

Reply:
I had started with a single function. However the parameterized test construction across 2 sets of tests repeats it causing a duplicate symbol linker error.

I've done some refactoring now to make the usage of these functions distinct.